### PR TITLE
APAC-4293: Add (GCP) metrics for relevant Redis backend, and broker methods

### PR DESCRIPTION
**Pull request by mattc-upguard**
While the second experiment should yield latency metrics from within the Redis server, these machinery metrics are intended to help determine whether the latency spikes are coming from machinery vs. Redis.

**Note: This should not actually be merged. The branch/commit just needs to exist (in GitHub) for webscan to depend on, and the PR is being made for review purposes only.**

### CODE REVIEW

File: v1/backends/redis/metrics.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+// It should also include any TagKeys that are needed
+// It must be registered inside init() above
+
+var machineryBackendRedisActionTimerMsDistView = &view.View{
+	Name:        mMachineryBackendRedisActionTimerMs.Name(),
+	Measure:     mMachineryBackendRedisActionTimerMs,
+	Description: mMachineryBackendRedisActionTimerMs.Description(),
+	TagKeys:     []tag.Key{kAction},
+	Aggregation: metrics.ExponentialTimerBucketsMs(100*time.Millisecond, 5*time.Second, 1.3),
+}
```
Comment (donners) [inline v1/backends/redis/metrics.go:54]: Might be good to also have a count of each call? You can create a "count" view of the same measure like I did [here](https://github.com/ScriptRock/vendor_risk/blob/14bf7d0707205f8fa7955f8d234cc75c99a7d137/ugdb/metrics.go#L35-L47).

**Comment by donners** [comment]
Looks good 👍 

Don't worry about metrics cost, I thought you were planning to leave them on indefinitely but if it's just for the experiment it's nothing to worry about.

**Review by donners — APPROVED** [review]

File: v1/backends/redis/metrics.go (hunk 1 above)
Comment (mattc-upguard) [reply v1/backends/redis/metrics.go:54]: Unless I'm misinterpreting the data, I think you get the count "for free". Looking at [this](https://console.cloud.google.com/monitoring/metrics-explorer;duration=P1D?project=vendor-risk-development&pageState=%7B%22xyChart%22:%7B%22constantLines%22:%5B%5D,%22dataSets%22:%5B%7B%22plotType%22:%22HEATMAP%22,%22targetAxis%22:%22Y1%22,%22timeSeriesFilter%22:%7B%22aggregations%22:%5B%7B%22crossSeriesReducer%22:%22REDUCE_SUM%22,%22groupByFields%22:%5B%5D,%22perSeriesAligner%22:%22ALIGN_DELTA%22%7D%5D,%22apiSource%22:%22DEFAULT_CLOUD%22,%22crossSeriesReducer%22:%22REDUCE_SUM%22,%22filter%22:%22metric.type%3D%5C%22custom.googleapis.com%2Fopencensus%2Fwebscan%2Fschedulers%2Fpriority%2Fqueueing_task_timer_ms%5C%22%20resource.type%3D%5C%22gke_container%5C%22%20resource.label.%5C%22project_id%5C%22%3D%5C%22vendor-risk-production%5C%22%20metric.label.%5C%22priority%5C%22%3D%5C%221%5C%22%20metric.label.%5C%22priority_scheduler_task%5C%22%3D%5C%22send%20chord%5C%22%22,%22groupByFields%22:%5B%5D,%22minAlignmentPeriod%22:%2260s%22,%22perSeriesAligner%22:%22ALIGN_DELTA%22%7D%7D%5D,%22options%22:%7B%22mode%22:%22COLOR%22%7D,%22y1Axis%22:%7B%22label%22:%22%22,%22scale%22:%22LINEAR%22%7D%7D%7D), I think it's the colour along the bottom. Do you think that will suffice?

**Comment by mattc-upguard** [comment]
> Don't worry about metrics cost, I thought you were planning to leave them on indefinitely but if it's just for the experiment it's nothing to worry about.

😌 

We could leave this branch around so that if we ever need the code again it's relatively accessible.

File: v1/backends/redis/metrics.go (hunk 1 above)
Comment (donners) [reply v1/backends/redis/metrics.go:54]: I find that pretty hard to read, but yea might be enough. Seems like there could be a way to get a total count out of that metric as a line chart, but I'm not sure.

**Comment by mattc-upguard** [comment]
Shouldn't hopefully need this again any time soon.

**Pull request closed without merging**